File: frontend/lib/models/category.dart
/// Beauty category (backend `categories.Category`).
class Category {
  const Category({
    required this.id,
    required this.slug,
    required this.name,
    this.description = '',
    this.iconMediaId,
    this.imageUrl = '',
    this.displayOrder = 0,
    this.isActive = true,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'] as String? ?? '',
      slug: json['slug'] as String? ?? '',
      name: json['name'] as String? ?? '',
      description: json['description'] as String? ?? '',
      iconMediaId: json['icon_media_id'] as String?,
      imageUrl: json['image_url'] as String? ?? '',
      displayOrder: (json['display_order'] as num?)?.toInt() ?? 0,
      isActive: json['is_active'] as bool? ?? true,
    );
  }

  final String id;
  final String slug;
  final String name;
  final String description;
  final String? iconMediaId;
  final String imageUrl;
  final int displayOrder;
  final bool isActive;

  bool get hasImage => imageUrl.isNotEmpty;
}

File: internal/categories/category.go
package categories

import "time"

type Category struct {
	ID           string    `json:"id"`
	Slug         string    `json:"slug"`
	Name         string    `json:"name"`
	Description  string    `json:"description,omitempty"`
	IconMediaID  *string   `json:"icon_media_id,omitempty"`
	ImageURL     string    `json:"image_url,omitempty"`
	DisplayOrder int       `json:"display_order"`
	IsActive     bool      `json:"is_active"`
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}
